superduperdb/ext/sentence_transformers/model.py

```diff
@@ -100,3 +97,22 @@ def predict(self, dataset: t.Union[t.List, QueryDataset]) -> t.List:
         if self.postprocess is not None:
             results = self.postprocess(results)  # type: ignore[operator]
         return results
+
+    def pre_create(self, db):
+        """Pre creates the model.
+
+        If the datatype is not set and the datalayer is an IbisDataBackend,
+        the datatype is set to ``sqlvector`` or ``vector``.
+
+        :param db: The datalayer instance.
+        """
+        super().pre_create(db)
+        if self.datatype is not None:
+            return
+
+        from superduperdb.components.vector_index import sqlvector, vector
+
+        if db.databackend.db_type == DBType.SQL:
```

Comment:
this logic is only required in sentence transformers?

Reply:
Other embedding models also have this logic.

We should consider merging `vector` and `sqlvector` into a single datatype later on. This should be handled at the underlying level and use the same datatype at the application layer.